src/track/track.cpp

```diff
@@ -227,10 +228,10 @@ void Track::importMetadata(
     }
 }
 
-bool Track::mergeImportedMetadata(
+bool Track::mergeExtraMetadataFromSource(
```

Comment:
From the naming I'd assume this only imports metadata that is not yet present (or empty). Is that correct?

Reply:
Not quite. It only affects metadata that is **not yet** stored in the database -> "extra"

We even have a corresponding test to verify that empty or missing fields like *artist* or *genre* are not modified.